### ISSUE TITLE
Question on F measure computation

### ISSUE THREAD

**Issue by tding1** [dnjs3594/Eigencontours#5]
Hi,

It seems that the F measure is computed by using `batch_f_measure` provided by davisinteractive:

https://github.com/dnjs3594/Eigencontours/blob/24be95d331a5a8611d1b1a0f98d1f555be7b796e/Preprocessing/code_v1_COCO/libs/S3_convert.py#L107-L111

But when I run the code, my F measure results are really high. For `M=8`, the average F score on COCO2017 of my experiments is around 0.84, which is much higher than that of around 0.4 reported in the paper (see Fig. 8 (c)).

In particular, even with `M=1`, the metric is very high. As a specific example, the following display imgs (left is the approximate mask with `cfg.dim=1`, right is the ground-truth mask) have a computed `batch_f_measure` of 1.0. I think there is something wrong since the two masks are so different. 

![image](https://user-images.githubusercontent.com/30458935/182052327-db4466f5-738e-4193-9243-399c53e6953d.png)

More related questions:

1. I take the SVD of the full contour matrix computed over COCO2017 train, whose size is `360 x 557905`, and during the convert step, I apply it to COCO2017  **val**. Although the paper does not mention how to use train/val, I assume this is the correct way. Can you please confirm this?

2. The paper mentions to compute the F measure, "bipartite matching is performed between the boundary points of a ground-truth contour and its approximated version" and then "F score is defined as the harmonic mean of the precision (P) and the recall (R) of the matching results". In the code this is done by using `batch_f_measure` provided by davisinteractive between the ground truth **masks** and approximate **masks**. Is this equivalent to the way described in the paper? 

Looking forward to your reply. Thank you very much!

**Comment by tding1** [comment]
I find that the parameter `bound_th` has a significant impact on the final F measure result. For example, for the default value of `bound_th=0.008` in the code, I get an average F score of 0.84 for COCO2017 val with `M=8`. However, if I set `bound_th=0`, I get an average F score of 0.288 for COCO2017 val with `M=8`. In either case, I cannot reproduce the F measure as reported in the paper. Can you please say anything about how do you set the parameter?

**Comment by dnjs3594** [comment]
First, we use batch_f_measure with default setting.
Second, we perform SVD on training dataset.
Third, we compare approximated masks with star-convex masks served as ground-truth masks.

**Comment by tding1** [comment]
Thanks! Then, I have a follow-up question:

During the encoding stage, what IOU threshold are you using in order to keep the contour data computed from the training data? I saw there is a default threshold value of 0.85 in the code, with which the final contour matrix is of size `360 x 557905`. Can you confirm with this?

**Comment by dnjs3594** [comment]
The thresholding is optional detail. The SVD without the thresholding is the default mode
